wild/tests/integration_tests.rs

```diff
@@ -1468,61 +1653,81 @@ fn integration_test(
     let configs = parse_configs(&src_path(filename))
         .with_context(|| format!("Failed to parse test parameters from `{filename}`"))?;
 
-    for config in configs {
-        if !config
-            .support_architectures
-            .contains(&get_host_architecture())
-            || config.requires_glibc && !cfg!(target_env = "gnu")
-            || (config.requires_clang_with_tlsdesc && !host_supports_clang_with_tls_desc())
-        {
-            eprintln!("Skipping config: {}", config.name);
-            continue;
-        }
-        let programs = linkers
-            .iter()
-            .filter(|linker| config.is_linker_enabled(linker))
-            .map(|linker| {
-                let start = Instant::now();
-                let result = program_inputs.build(linker, &config).with_context(|| {
-                    format!(
-                        "Failed to build program `{program_inputs}` \
+    let host_arch = get_host_architecture();
+
+    // We only currently support cross compilation from x86_64 to aarch64, so we don't need to track
+    // which targets are enabled, since there's only one.
+    let cross_enabled = std::env::var("WILD_TEST_CROSS").is_ok_and(|v| v == "aarch64");
+
+    for mut config in configs {
+        for &arch in &config.support_architectures {
+            let cross_arch = (arch != host_arch).then_some(arch);
+
+            if config.requires_glibc && !cfg!(target_env = "gnu")
+                || (config.requires_clang_with_tlsdesc && !host_supports_clang_with_tls_desc())
+                || (cross_arch.is_some()
+                    && (config.compiler == "clang" || !config.cross_enabled || !cross_enabled))
+            {
+                eprintln!(
+                    "Skipping config: {} for arch {}",
+                    config.name,
+                    cross_arch.map(|arch| arch.name()).unwrap_or("host")
+                );
+                continue;
+            }
+
+            if cross_arch.is_some() {
+                config.enabled_linkers.remove("lld");
+            }
```

Comment:
Is it because of test failures on Ubuntu due to output differences?
LLD should work fine in tests if accompanied by GNU cross toolchain.

Reply:
I think I hit some problems before. I can't remember what they were, but the tests still pass with this removed, so I've removed it.